vkcube (KHR_display) fails to build since v1.2.173 (-Werror, Clang, -DNDEBUG)

Regressed by 27c21248221f. Found [downstream](http://www.ipv6proxy.net/go.php?u=http://package23.nyi.freebsd.org/data/131i386-default-foo/2022-10-02_07h26m38s/logs/errors/vulkan-tools-1.3.230.log) because `-DCMAKE_BUILD_TYPE=Release` [adds `-DNDEBUG`](https://github.com/Kitware/CMake/blob/v3.24.0/Modules/Compiler/GNU.cmake#L59) thus makes `assert()` [expand to nothing](https://github.com/freebsd/freebsd-src/blob/releng/13.1/include/assert.h#L51).

```
$ cc --version
FreeBSD clang version 14.0.5 (https://github.com/llvm/llvm-project.git llvmorg-14.0.5-0-gc12386ae247c)
Target: x86_64-unknown-freebsd14.0
Thread model: posix
InstalledDir: /usr/bin
```

```c
$ cmake -DCMAKE_BUILD_TYPE=Release -DCUBE_WSI_SELECTION=DISPLAY -B /tmp/vulkan-tools_build
$ cmake --build /tmp/vulkan-tools_build
[...]
cube/cube.c:3122:14: error: variable 'result' set but not used [-Werror,-Wunused-but-set-variable]
    VkResult result;
             ^
1 error generated.
```

Note, GCC doesn't trip `-Werror` despite `BUILD_WERROR` enabled by default:
```c
cube/cube.c: In function 'find_display_gpu':
cube/cube.c:3122:14: warning: variable 'result' set but not used [-Wunused-but-set-variable]
 3122 |     VkResult result;
      |              ^~~~~~
```

--- a/cube/cube.c
+++ b/cube/cube.c
@@ -3119,7 +3119,7 @@ static VkBool32 demo_check_layers(uint32_t check_count, char **check_names, uint
 #if defined(VK_USE_PLATFORM_DISPLAY_KHR)
 int find_display_gpu(int gpu_number, uint32_t gpu_count, VkPhysicalDevice *physical_devices) {
     uint32_t display_count = 0;
-    VkResult result;
+    VkResult U_ASSERT_ONLY result;
     int gpu_return = gpu_number;
     if (gpu_number >= 0) {
         result = vkGetPhysicalDeviceDisplayPropertiesKHR(physical_devices[gpu_number], &display_count, NULL);

